test-app/tests/integration/components/toucan-form/toucan-form-test.gts

```diff
@@ -119,12 +132,30 @@ module('Integration | Component | ToucanForm', function (hooks) {
     assert.dom('[data-checkbox-group-1]').isChecked();
     assert.dom('[data-checkbox-group-2]').isNotChecked();
     assert.dom('[data-checkbox-group-3]').isChecked();
+
+    // File input
+    assert.dom('[data-files] [data-file-name]').hasText('file.txt');
   });
 
   test('it triggers validation and shows error messages in the Toucan Core components', async function (assert) {
-    assert.expect(13);
+    assert.expect(19);
+
+    const handleSubmit = ({ files, ...data }: TestData) => {
+      // Checking deep equality on files can get really tricky due
+      // to properties on the File object prototype like `arrayBuffer, `lastModifiedDate`,
+      // and `lastModified`.  Due to that, we will assert `files` separately from the other
+      // submitted values.
+      assert.ok(files, 'Expected files to be included');
+      assert.strictEqual(
+        files?.length,
+        1,
+        'Expected only a single file to be added'
+      );
+      assert.strictEqual(files?.[0]?.name, 'file.txt');
```

Comment:
I'm not suggesting a change. Just opining. 

I've gotten into the habit of using `Array.prototype.at` for non-tuple array-index access. It produces the same result, of course, but `Array.prototype.at`'s return type includes `undefined`, which forces me to assert against `undefined` instead of possibly allowing a runtime error to propagate. 

Unicorn [has a rule](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-at.md) for it. Unfortunately, I can't find a one-off plugin with the rule.

Reply:
Though `FileList` is array-like and so doesn't support `.at`. It does support `.item`.

Reply:
> but Array.prototype.at's return type includes undefined, which forces me to assert against undefined instead of possibly allowing a runtime error to propagate.

That's a good point! That really is a pretty bad mistake by TS to not do this for normal index access. Do you happen to know if there are plans to change this? It would probably be the right thing to do, although it would likely cause a ton of new type errors. But TS doesn't follow Semver anyway, so... 

> Though FileList is array-like and so doesn't support .at. It does support .item.

IIRC we chose to pass the files as `File[]`, so it _is_ an array if I am not mistaken...

Reply:
> Do you happen to know if there are plans to change this?

It's already there ([since ts 4.1](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#checked-indexed-accesses---nouncheckedindexedaccess)) behind a config flag (`noUncheckedIndexedAccess`). I'm pretty sure the safer version is the default in the `tsconfig.json` files recommended by `ember` (we don't have it on internally but IIRC it is on in the v2 addon blueprint which has caused confusion when people migrate to v2 addons).

Reply:
> IIRC we chose to pass the files as File[], so it is an array if I am not mistaken...

Hm. If that's the reason, I wonder if we wouldn't be better off converting to an array as needed using `Array.from`.